### PR TITLE
Microsoft Purview Share Java SDK Changes for Version 2023-05-30-preview

### PR DESCRIPTION
**Pull request by kevinmichaelbowersox**
# Microsoft Purview Share Java SDK Version 2023-05-30-preview

This pull requests updates the Microsoft Purview Share Java SDK for a new version of the service. This new version of the service adds an additional capability for management of a ShareResource. A share resource is the underlying resource from which a provider shares data or the destination where a consumer attaches data shared with them. Here is a link to the new [OpenApi specification file](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/purview/data-plane/Azure.Analytics.Purview.Share/preview/2023-05-30-preview/share.json) and its [PR](https://github.com/Azure/azure-rest-api-specs/commit/40a953243ea428918de6e63758e853b7a24aa59a).

Specific changes in this pull request include:
- Regeneration of the SDK for the new version of the OpenApi Specification that includes the ShareResource changes. Generation of the SDK was performed with --enable-sync-stack=false per recommendations from the SDK team.
- README.md was improved to better describe the sharing flows performed by SDK clients.
- Samples and Tests were added for the new ShareResource capability.
- New test session records were recorded for ShareResource. A problematic/invalid UUID was addressed that caused existing tests to fail.
- Small changes to description in the pom.xml

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/purview/azure-analytics-purview-sharing/src/test/java/com/azure/analytics/purview/sharing/generated/PurviewShareClientTestBase.java [hunk 1]
```diff
@@ -61,5 +65,23 @@ protected void beforeTest() {
...
+
+        ShareResourcesClientBuilder shareResourcesClientbuilder =
+                new ShareResourcesClientBuilder()
+                        .endpoint(Configuration.getGlobalConfiguration().get("ENDPOINT", "endpoint"))
+                        .httpClient(HttpClient.createDefault())
+                        .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+        if (getTestMode() == TestMode.PLAYBACK) {
+            shareResourcesClientbuilder
+                    .httpClient(interceptorManager.getPlaybackClient())
+                    .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
```
Comment (samvaity) [inline sdk/purview/azure-analytics-purview-sharing/src/test/java/com/azure/analytics/purview/sharing/generated/PurviewShareClientTestBase.java:77]:
```suggestion
                    .credential(new MockTokenCredential());
```

File: sdk/purview/azure-analytics-purview-sharing/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Release History
 
-## 1.0.0-beta.2 (Unreleased)
+## 1.0.0-beta.2 (2023-05-30)
```
[the remark this answers is not in the document]
Comment (kevinmichaelbowersox) [reply sdk/purview/azure-analytics-purview-sharing/CHANGELOG.md:3]: Fixed in next update.

File: sdk/purview/azure-analytics-purview-sharing/src/test/java/com/azure/analytics/purview/sharing/ShareResourceClientTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.jupiter.api.Test;
+
+import com.azure.analytics.purview.sharing.models.InPlaceSentShare;
+import com.azure.analytics.purview.sharing.models.SentShare;
+import com.azure.analytics.purview.sharing.models.ShareResource;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+
+@SuppressWarnings("unused")
```
[the remark this answers is not in the document]
Comment (kevinmichaelbowersox) [reply sdk/purview/azure-analytics-purview-sharing/src/test/java/com/azure/analytics/purview/sharing/ShareResourceClientTest.java:19]: Removed in next update.

File: sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return a page of ShareResource results as paginated response with {@link PagedFlux}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<BinaryData> getAllShareResources(RequestOptions requestOptions) {
```
[the remark this answers is not in the document]
Comment (kevinmichaelbowersox) [reply sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java:76]: We have a transformation for this in place within the swagger/README.md to adjust the autorest generation.

File: sdk/purview/azure-analytics-purview-sharing/src/test/java/com/azure/analytics/purview/sharing/generated/PurviewShareClientTestBase.java (hunk 1 above)
Comment (kevinmichaelbowersox) [reply sdk/purview/azure-analytics-purview-sharing/src/test/java/com/azure/analytics/purview/sharing/generated/PurviewShareClientTestBase.java:77]: Fixed.

File: sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (samvaity) [reply sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java:76]: I see, that is also the pattern we have used across the lib too.
Although, I am not sure if we should. 
Was there a reason we wanted it named this pattern?

File: sdk/purview/azure-analytics-purview-sharing/swagger/README.md [hunk 5]
```diff
@@ -45,5 +45,7 @@ directive:
       to: "ReceivedShares_GetAllAttachedReceivedShares",
       from: "ReceivedShares_ListDetached",
       to: "ReceivedShares_GetAllDetachedReceivedShares",
+      from: "ShareResources_List",
+      to: "ShareResources_GetAllShareResources"
```
Comment (srnagar) [inline sdk/purview/azure-analytics-purview-sharing/swagger/README.md:49]: All operations that return a paged response should have `list` as the prefix in Java, Python and JS. Only .NET uses `Get` prefix for paged response.

File: sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (srnagar) [reply sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java:76]: We should name this `listAllShareResources()`. This is the naming guideline in Java for methods that return a paged response.

[the remark this answers is not in the document]
Comment (kevinmichaelbowersox) [reply sdk/purview/azure-analytics-purview-sharing/src/main/java/com/azure/analytics/purview/sharing/ShareResourcesAsyncClient.java]: Thank you both for the guidance.  I have updated our transformations to remove `GetAll` and to use `List` instead in all instances where `GetAll` was used.

File: sdk/purview/azure-analytics-purview-sharing/swagger/README.md (hunk 5 above)
Comment (kevinmichaelbowersox) [reply sdk/purview/azure-analytics-purview-sharing/swagger/README.md]: Thank you for the guidance.  I have updated our transformations to remove `GetAll` and to use `List` instead in all instances where `GetAll` was used.

File: sdk/purview/azure-analytics-purview-sharing/README.md [hunk 6]
```diff
@@ -130,55 +219,28 @@ SentSharesClient sentSharesClient =
...
-                sentShareId,
-                sentShareInvitationId,
-                BinaryData.fromObject(sentShareInvitation),
-                new RequestOptions());
+Response<BinaryData> sentShareInvitation =
+        sentSharesClient.getSentShareInvitationWithResponse(sentShareId, sentShareInvitationId, new RequestOptions());
 ```
 
-### Get All Sent Share Invitations
+#### Get All Sent Share Invitations
```
Comment (samvaity) [inline sdk/purview/azure-analytics-purview-sharing/README.md:240]: nit: List Sent Share Invitations
here and other places

**Review by samvaity — APPROVED** [review]
Changes lgtm!

**Pull request merged**